pkg/console/controllers/downloadsdeployment/controller.go

```diff
@@ -0,0 +1,139 @@
+		// events
+		resourceSyncer: resourceSyncer,
+	}
+
+	configNameFilter := util.NamesFilter(api.ConfigResourceName)
+	downloadsNameFilter := util.NamesFilter(api.OpenShiftConsoleDownloadsDeploymentName)
+
+	return factory.New().
+		WithFilteredEventsInformers( // infrastructure configs
+			configNameFilter,
+			operatorConfigInformer.Informer(),
+			configV1Informers.Infrastructures().Informer(),
+		).WithFilteredEventsInformers( // downloads deployment
+		downloadsNameFilter,
+		deploymentInformer.Informer(),
+	).WithSync(ctrl.Sync).
+		ToController("ConsoleDownloadsDeploymentSyncController", recorder.WithComponentSuffix("console-downloads-deployment-controller"))
+}
+
+func (c *DownloadsDeploymentSyncController) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	operatorConfig, err := c.operatorConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+
+	switch updatedOperatorConfig.Spec.ManagementState {
+	case operatorv1.Managed:
+		klog.V(4).Infoln("console is in a managed state: syncing downloads deployment")
+	case operatorv1.Unmanaged:
+		klog.V(4).Infoln("console is in an unmanaged state: skipping downloads deployment sync")
+		return nil
+	case operatorv1.Removed:
+		klog.V(4).Infoln("console is in an removed state: removing synced downloads deployment")
+		return c.removeDownloadsDeployment(ctx)
+	default:
+		return fmt.Errorf("unknown state: %v", updatedOperatorConfig.Spec.ManagementState)
+	}
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	infrastructureConfig, err := c.infrastructureConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	statusHandler.AddCondition(status.HandleDegraded("DonwloadsDeploymentSync", "FailedInfrastructureConfigGet", err))
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	downloadsDeploymentErr := c.SyncDownloadsDeployment(ctx, updatedOperatorConfig, infrastructureConfig, controllerContext)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("DownloadsDeploymentSync", "FailedApply", downloadsDeploymentErr))
+	if downloadsDeploymentErr != nil {
+		return statusHandler.FlushAndReturn(downloadsDeploymentErr)
+	}
+
+	return statusHandler.FlushAndReturn(nil)
+}
+
+func (c *DownloadsDeploymentSyncController) SyncDownloadsDeployment(ctx context.Context, operatorConfig *operatorv1.Console, infrastructureConfig *configv1.Infrastructure, controllerContext factory.SyncContext) error {
+
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+	requiredDownloadsDeployment := deploymentsub.DefaultDownloadsDeployment(operatorConfig, infrastructureConfig)
+
+	_, _, applyErr := resourceapply.ApplyDeployment(
+		c.deploymentClient,
+		controllerContext.Recorder(),
+		requiredDownloadsDeployment,
+		resourcemerge.ExpectedDeploymentGeneration(requiredDownloadsDeployment, updatedOperatorConfig.Status.Generations),
+	)
```

Comment:
Should there be more detailed conditions describing the status of this deployment? Is this done elsewhere?

Reply:
yes, when syncing `console` deployment.
https://github.com/openshift/console-operator/blob/master/pkg/console/operator/sync_v400.go#L247